tests/functional/test_sessions_mub.py

```diff
@@ -0,0 +1,303 @@
+import pytest
+from pydantic_marshals.contains import TypeChecker
+from starlette.testclient import TestClient
+
+from app.models.sessions_db import Session
+from app.models.users_db import User
+from app.utils.authorization import AUTH_COOKIE_NAME
+from tests.conftest import ActiveSession, Factory
+from tests.mock_stack import MockStack
+from tests.utils import assert_nodata_response, assert_response, get_db_user
+
+
+def mub_session_checker(session: Session, invalid: bool = False) -> TypeChecker:
+    return {
+        "mub": session.mub,
+        "id": session.id,
+        "created": session.created.isoformat(),
+        "expiry": session.expiry.isoformat(),
+        "disabled": invalid,
+        "invalid": invalid,
+    }
+
+
+@pytest.mark.anyio()
+async def test_making_mub_session(
+    client: TestClient,
+    user: User,
+) -> None:
+    assert_nodata_response(
+        client.post(f"/mub/users/{user.id}/sessions/"),
+        expected_code=204,
+    )
+
+
+@pytest.mark.anyio()
+async def test_taking_mub_session(
+    client: TestClient,
+    user: User,
+    active_session: ActiveSession,
+) -> None:
+    async with active_session():
+        await Session.create(user_id=(await get_db_user(user)).id, mub=True)
+
+    assert_nodata_response(
+        client.put(f"/mub/users/{user.id}/sessions/"),
+        expected_code=204,
+    )
+
+
+@pytest.mark.anyio()
+async def test_getting_all_sessions(
+    client: TestClient,
+    user: User,
+    session_factory: Factory[Session],
+    active_session: ActiveSession,
+) -> None:
+    async with active_session():
+        sessions = [await session_factory()]
+    assert_response(
+        client.get(f"/mub/users/{user.id}/sessions/"),
+        expected_json=[mub_session_checker(session) for session in sessions],
+    )
+
+
+@pytest.mark.anyio()
+async def test_disabling_session(
+    client: TestClient,
+    user: User,
+    session: Session,
+    active_session: ActiveSession,
+) -> None:
+    assert_nodata_response(
+        client.delete(f"/mub/users/{user.id}/sessions/{session.id}/"),
+        expected_code=204,
+    )
+    async with active_session():
+        db_session = await Session.find_first_by_id(session.id)
+        assert db_session is not None
+        assert db_session.invalid
+
+
+@pytest.mark.anyio()
+async def test_deleting_session(
+    client: TestClient,
+    user: User,
+    session: Session,
+    active_session: ActiveSession,
+) -> None:
+    assert_nodata_response(
+        client.delete(
+            f"/mub/users/{user.id}/sessions/{session.id}/",
+            params={"delete_session": "true"},
+        ),
+        expected_code=204,
+    )
+    async with active_session():
+        db_session = await Session.find_first_by_id(session.id)
+        assert db_session is None
```

Comment:
fix: можно в одну строчку написать, еще пару таких случаев видел внизу
```suggestion
        assert (await Session.find_first_by_id(session.id)) is None
```

Reply:
А если это несколько случаев, может можно вынести, как вынесли `get_db_user`?